drivers/include/sha5461as.h

```diff
@@ -0,0 +1,158 @@
+    gpio_t dig2;    /**< Data input pin */
+    gpio_t dig3;    /**< Data input pin */
+    gpio_t dig4;    /**< Data input pin */
+} sha5461as_params_t;
+
+/**
+ * @brief   Device descriptor for the SHA5461AS
+ */
+typedef struct {
+    sha5461as_params_t params;  /**< Device parameters */
+    uint32_t value;             /**< Binary value to display */
+    uint8_t digits;             /**< Number of digits */
+    uint8_t current_digit;      /**< Displayed digit */
+    int status;                 /**< Status of the timer_periodic for this dev */
+} sha5461as_t;
+
+/**
+ * @brief   Initialize the given DSP0401
+ *
+ * @param[out] dev      Initialized device descriptor of SHA5461AS device
+ * @param[in]  params   Device parameters to use
+ *
+ * @return              SHA5461AS_OK if everything is good
+ * @return              -SHA5461AS_ERR_A_GPIO if an error occurred during A GPIO
+ * @return              -SHA5461AS_ERR_B_GPIO if an error occurred during B GPIO
+ * @return              -SHA5461AS_ERR_C_GPIO if an error occurred during C GPIO
+ * @return              -SHA5461AS_ERR_D_GPIO if an error occurred during D GPIO
+ * @return              -SHA5461AS_ERR_E_GPIO if an error occurred during E GPIO
+ * @return              -SHA5461AS_ERR_F_GPIO if an error occurred during F GPIO
+ * @return              -SHA5461AS_ERR_G_GPIO if an error occurred during G GPIO
+ * @return              -SHA5461AS_ERR_DP_GPIO if an error occurred during DP GPIO
+ * @return              -SHA5461AS_ERR_DIG1_GPIO if an error occurred during DIG1 GPIO
+ * @return              -SHA5461AS_ERR_DIG2_GPIO if an error occurred during DIG2 GPIO
+ * @return              -SHA5461AS_ERR_DIG3_GPIO if an error occurred during DIG3 GPIO
+ * @return              -SHA5461AS_ERR_DIG4_GPIO if an error occurred during DIG4 GPIO
+ */
+int sha5461as_init(sha5461as_t *dev, const sha5461as_params_t *params);
+
+/**
+ * @brief   Shift the current digit off and the next one on
+ *
+ * @param[in] dev       Initialized device descriptor of SHA5461AS device
+ *
+ */
+int sha5461as_shift(sha5461as_t *dev);
+
+/**
+ * @brief   Set the value for every digit
+ *
+ * @param[in] dev       Initialized device descriptor of SHA5461AS device
+ * @param[in] value     the value as an uint32_t
+ *
+ */
+int sha5461as_set_all_value(sha5461as_t *dev, uint32_t value);
+
+/**
+ * @brief   Set the value for one digit
+ *
+ * @param[in] dev       Initialized device descriptor of SHA5461AS device
+ * @param[in] index     the digit number between [0, SHA5461AS_DIGITS[
+ * @param[in] value     the value as an uint8_t, A is equal to the first bit
+ *                        B the second.... DP the 8th bit.
+ *
+ */
+int sha5461as_set_value(sha5461as_t *dev, int index, uint8_t value);
+
+/**
+ * @brief   Start/stop an periodic timer event to shift between each 7seg
+ *
+ * @param[in] dev       Initialized device descriptor of SHA5461AS device
+ */
+int sha5461as_display(sha5461as_t *dev);
```

Comment:
The usual way is to provide a set of `DRIVERNAME_poweron()` and `DRIVERNAME_poweron()` functions rather than a single function that toggles the hardware on/off.